### ISSUE TITLE
Not work Cal and CardDav sync with DavDroid after upgrade to OC9

### ISSUE THREAD

**Issue by Loki3000** [owncloud/core#23147]
### Steps to reproduce
1. trying to add new account in DAVdroid
2. get error '401 Unauthorized'
([client log](https://github.com/owncloud/core/files/168928/davdroid_401.txt))
3. check log at server: get error '401 CSRF check not passed' ([server log](https://github.com/owncloud/core/files/168933/oc_401.txt)). Possible it's DAVdroid problem, so i comment following check in apps/dav/lib/connector/sabre/auth.php
```php
if(!$this->request->passesCSRFCheck()) {
    $response->setStatus(401);
    throw new \Sabre\DAV\Exception\NotAuthenticated('CSRF check not passed.');
}
```
4. now account adds properly, but while synchronization get error '404 Addressbook with name 'contacts' could not be found': [client log](https://github.com/owncloud/core/files/168927/davdroid_404.txt), [server log](https://github.com/owncloud/core/files/168926/oc_404.txt)

5. I check path in log files and comare it's with path from web client. It's same as I see (https://mydomain/remote.php/dav/addressbooks/users/user/contacts)

### Expected behaviour
synchronization account
### Actual behaviour
Getting errors described above.

### Server configuration
**Operating system**: Ubuntu 14.04
**Web server:** Apache/2.4.18
**Database:** mysql  5.5.47
**PHP version:** PHP 5.5.9
**ownCloud version:** ownCloud 9.0.0 (stable)
**Updated from an older ownCloud or fresh install:** updated from 8.2
**Where did you install ownCloud from:** repository
**Signing status (ownCloud 9.0 and above):**

```
http://example.com/index.php/settings/integrity/failed 
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_mv: 0.8.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - news: 7.1.2
  - notes: 2.0.0
  - notifications: 0.2.3
  - ocdownloader: 1.5
  - provisioning_api: 0.4.1
  - rainloop: 4.12
  - search_lucene: 0.7.0
  - systemtags: 0.2
  - tasks: 0.9.0
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - updater: 0.6
Disabled:
  - direct_menu
  - encryption
  - external
  - user_external
  - user_ldap

```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "xxx",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "<mydomain>"
        ],
        "datadirectory": "\/home\/cloud\/data",
        "overwrite.cli.url": "https:\/\/<mydomain>",
        "logfile": "\/var\/log\/owncloud.log",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "db_cloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 0,
        "theme": "",
        "maintenance": false,
        "mail_from_address": "root",
        "mail_smtpmode": "php",
        "mail_domain": "mail-domain",
        "forcessl": false,
        "trashbin_retention_obligation": "auto",
        "appstore.experimental.enabled": true,
        "defaultapp": "rainloop",
        "filesystem_check_changes": 1,
        "asset-pipeline.enabled": false,
        "updatechecker": false,
        "logtimezone": "UTC"
    }
}

```

**Are you using external storage, if yes which one:** dropbox

**Are you using encryption:** no

### Client configuration
**Browser:** DAVdroid version: 0.9.1.3

**Operating system:** Android 4.4.2

**Comment by reuterbal** [comment]
Probably the same as #22988

**Comment by Loki3000** [comment]
Seems that I found reason. Issue that Auth is case insensitive, but ACL is case sensitive.
As a result in file
3rdparty/sabre/dav/lib/DAVACL/Plugin.php
happened follow:
https://domain/remote.php/dav/addressbooks/users/User/contacts/
User:password

```php
$principals = $this->getCurrentUserPrincipals();
//returns follows:
Array
(
    [0] => principals/users/User
)
```
and for user:password it will be that:
```php
$principals = $this->getCurrentUserPrincipals();
//returns follows:
Array
(
    [0] => principals/users/user
)
```
as a result, condition 
```php
//$principals='principals/users/User' - it gets from URL
if (in_array($ace['principal'], $principals)) {
    $collected[] = $ace;
}
```
is never met and $collected always empty. As a result, we get error 404.


PS I still don't know why CSRF check falls.

**Comment by LukasReschke** [comment]
cc @DeepDiver1975

**Comment by LukasReschke** [comment]
> PS I still don't know why CSRF check falls.

I don't have an Android phone myself but from what I can see in your logs there are authentication requests done at the wrong endpoint. Would it be possible for you to attach a proxy and share the sniffed traffic with me? (you can find my mail address in my profile)

**Comment by Loki3000** [comment]
@LukasReschke, I send dump to your email.

**Comment by LukasReschke** [comment]
@Loki3000 Thanks a lot. Judging from the dump I'd say the CSRF error is fixed with https://github.com/owncloud/core/pull/23237. Could you test this?

Probably the best way to test this would be to download https://download.owncloud.org/community/owncloud-daily-stable9.tar.bz2 as this also contains other DAV related patches.

**Comment by Loki3000** [comment]
> CSRF error is fixed with #23237. Could you test this?
I apply this patch to stable release and it's work well. Thanks!

**Comment by LukasReschke** [comment]
@Loki3000 Thanks a lot. Awesome.

Can you open a new issue if any other of the issues listed in the top post are still applicable? Thanks a lot! :)
